test/run-tests.sh

```diff
@@ -130,32 +149,54 @@ exprForCI() {
     ((memAvailableMiB < memoryMiB)) && memoryMiB=$memAvailableMiB
     >&2 echo "VM stats: CPUs: $numCPUs, memory: $memoryMiB MiB"
     >&2 echo "Host memory total: $((memTotalKiB / 1024)) MiB, available: $memAvailableMiB MiB"
-    vmTestNixExpr
+
+    # VMX is usually not available on CI nodes due to recursive virtualisation.
+    # Explicitly disable VMX, otherwise QEMU 4.20 fails with message
+    # "error: failed to set MSR 0x48b to 0x159ff00000000"
+    vmTestNixExpr "-cpu host,-vmx"
 }
 
 vmTestNixExpr() {
+  extraQEMUOpts="$1"
   cat <<EOF
-    (import "$scriptDir/test.nix" { scenario = "$scenario"; } {}).overrideAttrs (old: rec {
+    ((import "$testDir/tests.nix" { scenario = "$scenario"; }).vm {}).overrideAttrs (old: rec {
       buildCommand = ''
-        export QEMU_OPTS="-smp $numCPUs -m $memoryMiB"
+        export QEMU_OPTS="-smp $numCPUs -m $memoryMiB $extraQEMUOpts"
         echo "VM stats: CPUs: $numCPUs, memory: $memoryMiB MiB"
       '' + old.buildCommand;
     })
 EOF
 }
 
+# A basic subset of tests to keep the total runtime within
+# manageable bounds (<3 min on desktop systems).
+# These are also run on the CI server.
+basic() {
+    scenario=default buildTest "$@"
+    scenario=netns buildTest "$@"
+    scenario=full evalTest "$@"
+}
+
+all() {
+    scenario=default buildTest "$@"
+    scenario=netns buildTest "$@"
+    scenario=full buildTest "$@"
+}
```

Comment:
2 min 30 vs 3 min 15 on my laptop so we can switch to all by default in the future if we miss issues with the full test.